quic/s2n-quic-core/src/datagram.rs

```diff
@@ -1,18 +1,38 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
+
+/// The datagram endpoint trait provides a way to implement custom unreliable datagram
+/// sending and receiving logic. The Sender type should be implemented for custom
+/// sending behavior, and the Receiver type should be implemented for custom
+/// receiving behavior.
 pub trait Endpoint: 'static + Send {
     type Sender: Sender;
     type Receiver: Receiver;
 
-    fn create_connection(&mut self, info: &ConnectionInfo) -> (Self::Sender, Self::Receiver);
+    fn split(&mut self) -> (Self::Sender, Self::Receiver);
```

Comment:
I know we've been spinning on this name but I'm not sure `split` is better than what's there. We're not splitting the endpoint; we're creating an instance of a sender and receiver for a new connection. Also we should continue to have `&ConnectionInfo` so the endpoint can decide how to create the instance.

Reply:
Do you have any thoughts on what should be inside of ConnectionInfo?

Reply:
we can leave it blank for now. but the idea is to reserve this struct for future needs. (examples could include application protocol, server name, etc)

Reply:
> keep ConnectionInfo for later

can you add a comment saying why its there. seems like its going to be a common question.